container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/engine_functions/destroy_engines.go

```diff
@@ -61,5 +62,22 @@ func DestroyEngines(
 		)
 	}
 
+	// This is a small hack so that the log aggregator isn't cleaned while trying to remove stopped engines (eg. kurtosis clean -a)
+	shouldRemoveLogComponents := len(matchingEnginesByContainerId) == 0
+	if shouldRemoveLogComponents {
+		if err := removeCentralizedLogsComponents(ctx, dockerManager); err != nil {
+			return nil, nil, stacktrace.Propagate(err, "An error occurred removing the logging components.")
+		}
+	}
+
 	return successfulGuids, erroredGuids, nil
 }
+
+// ====================================================================================================
+//
+//	Private helper methods
+//
+// ====================================================================================================
+func removeCentralizedLogsComponents(ctx context.Context, dockerManager *docker_manager.DockerManager) error {
```

Comment:
Nit: you can inline this call directly above. Saves you the energy of creating a function